### PR TITLE
po4a: fix build on Big Sur and bump to version 0.62

### PR DESCRIPTION
**Pull request by deatondg**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
There is one for a version bump, but it is still broken on Big Sur. [https://github.com/Homebrew/homebrew-core/pull/66522](https://github.com/Homebrew/homebrew-core/pull/66522)
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

As in issue #67333, po4a is completely broken on Big Sur. 

The first reason is that "Term::ReadKey" fails to build with system perl. Consequently, I have switched to Homebrew perl, and the build succeeds. Perhaps someone can figure out a way around this, but I think requiring Homebrew perl is a better solution than shipping a segfault'ing program.

The second reason is that the po4a build script was unnecessarily reseting the PERL5LIB environment variable. My PR on po4a to fix this just got accepted, so this issue will not be relevant in the next version of po4a. I have still included the patch here, but it can be removed soon.

In summary:
Changes:

- Version bump to 0.62
- The fix from https://github.com/Homebrew/homebrew-core/pull/66522
- Switch to Homebrew perl
- Include small patch to po4a

Fixes issue #67333

### CODE REVIEW

**Comment by deatondg** [comment]
It was bugging me that we had to switch to Homebrew perl, so I spent some time looking into why Term::ReadKey doesn't build on system perl. Turns out it's exactly the reason discussed in chenrui333's PR and I just should have read better. 

After a two character change, chenrui333's fix works correctly on Big Sur, and we do not need to switch to Homebrew perl (or use my fix to po4a, for some reason). 

I chose to rename the Term::ReadKey resource name to TermReadKey instead of changing the if statement to Term::ReadKey because the latter resulted in some stupid Makefile errors due to `:`s in paths.

**Comment by mitchblank** [comment]
Ah that makes sense... @chenrui333 was trying to adapt my change from #66699 and I was surprised to hear it hadn't worked.  I meant to do some debugging myself but hadn't gotten around to it.  You figured it out with the `if` mistake

**Review by chenrui333 — APPROVED** [review]
Nice work. Thanks!

**Comment by mitchblank** [comment]
@chenrui333 -- can you also kick off a rebottling of `dpkg`?  I think the only reason that hadn't bottled on Big Sur earlier was that `po4a` was missing
